src/ApparentHorizons/Strahlkorper.cpp

```diff
@@ -0,0 +1,126 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "ApparentHorizons/Strahlkorper.hpp"
+
+#include "ApparentHorizons/SpherepackIterator.hpp"
+#include "ErrorHandling/Assert.hpp"
+#include "Parallel/PupStlCpp11.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+
+template <typename Frame>
+Strahlkorper<Frame>::Strahlkorper(const size_t l_max, const size_t m_max,
+                                  const double radius,
+                                  std::array<double, 3> center) noexcept
+    : l_max_(l_max),
+      m_max_(m_max),
+      ylm_(l_max, m_max),
+      // clang-tidy: do not std::move trivially constructable types
+      center_(std::move(center)),  // NOLINT
+      strahlkorper_coefs_(ylm_.spectral_size(), 0.0) {
+  ylm_.add_constant(&strahlkorper_coefs_, radius);
+}
+
+template <typename Frame>
+Strahlkorper<Frame>::Strahlkorper(
+    const size_t l_max, const size_t m_max,
+    const Strahlkorper& another_strahlkorper) noexcept
+    : l_max_(l_max),
+      m_max_(m_max),
+      ylm_(l_max, m_max),
+      center_(another_strahlkorper.center_),
+      strahlkorper_coefs_(another_strahlkorper.ylm_.prolong_or_restrict(
+          another_strahlkorper.strahlkorper_coefs_, ylm_)) {}
+
+template <typename Frame>
+Strahlkorper<Frame>::Strahlkorper(const Strahlkorper& another_strahlkorper,
+                                  DataVector coefs) noexcept
+    : l_max_(another_strahlkorper.l_max_),
+      m_max_(another_strahlkorper.m_max_),
+      ylm_(another_strahlkorper.ylm_),
+      center_(another_strahlkorper.center_),
+      strahlkorper_coefs_(std::move(coefs)) {
+  ASSERT(
+      strahlkorper_coefs_.size() == another_strahlkorper.ylm_.spectral_size(),
+      "Bad size " << strahlkorper_coefs_.size() << ", expected "
+                  << another_strahlkorper.ylm_.spectral_size());
+}
+
+template <typename Frame>
+Strahlkorper<Frame>::Strahlkorper(
+    const DataVector& radius_at_collocation_points, const size_t l_max,
+    const size_t m_max, std::array<double, 3> center) noexcept
+    : l_max_(l_max),
+      m_max_(m_max),
+      ylm_(l_max, m_max),
+      // clang-tidy: do not std::move trivially constructable types
+      center_(std::move(center)),  // NOLINT
+      strahlkorper_coefs_(ylm_.phys_to_spec(radius_at_collocation_points)) {
+  ASSERT(radius_at_collocation_points.size() == ylm_.physical_size(),
+         "Bad size " << radius_at_collocation_points.size() << ", expected "
+                     << ylm_.physical_size());
+}
+
+template <typename Frame>
+void Strahlkorper<Frame>::pup(PUP::er& p) {
+  p | l_max_;
+  p | m_max_;
+  p | center_;
+  p | strahlkorper_coefs_;
+
+  if (p.isUnpacking()) {
+    ylm_ = YlmSpherepack(l_max_, m_max_);
+  }
+}
+
+template <typename Frame>
+std::array<double, 3> Strahlkorper<Frame>::physical_center() const noexcept {
+  // Uses Eqs. (38)-(40) in Hemberger et al, arXiv:1211.6079.  This is
```

Comment:
could you have the arXiv text be a hyperlink?

Reply:
This is not a doxygen comment.  Should it be?  (If not, I don't know how to hyperlink)

Reply:
Whoops! I think this is fine. Could you just link the paper in the Doxygen comment that's already in the header?